Added checkbox to link modal.

JoshSinghONS opened the pull request:
What
Added checkbox so that if a user checks it the code will lop off the domain portion from the string via regex.

How to review
Look at the code.
This branch cannot be tested as there is no dev environment however a working local version can be found via the florence repo and the branch fix/wip-josh

Who can review
Anyone.

AnnaEmson reviewed (commented):
Hi
I looked at your PR. There is no info provided. Maybe some screenshots or video would speed up the review process :slightly_smiling_face: Probably some instructions how to test it would be handy too :slightly_smiling_face:
But anyway, we can read code, so please correct me if I am wrong:
you added new input checkbox to the page with 'Internal' label and some styling
when checked it is stripping the domain part from url to something like '/boo/foo'.
I am just wondering how it works, as checkboxes value is sent to the server when the form submission happens, it is not available to front-end? I haven't work with that part of code so sorry if I am asking obvious questions.  What happen when it will be toggled?  I know I am usually on front end rely on checked attribute when showing state ?

JoshSinghONS commented:
> Hi
> I looked at your PR. There is no info provided. Maybe some screenshots or video would speed up the review process 🙂 Probably some instructions how to test it would be handy too 🙂
> But anyway, we can read code, so please correct me if I am wrong:
> you added new input checkbox to the page with 'Internal' label and some styling
> when checked it is stripping the domain part from url to something like '/boo/foo'.
> I am just wondering how it works, as checkboxes value is sent to the server when the form submission happens, it is not available to front-end? I haven't work with that part of code so sorry if I am asking obvious questions. What happen when it will be toggled? I know I am usually on front end rely on checked attribute when showing state ?

Hello,

So your right in your assumption of the change, added check box with styling and a conditional checking for said check box value.

In terms of how it works this is actually my first time working with the codebase so I'm not sure of the entire flow of the form either lol but looking at the code the onSubmit goes to a function called close() first, which is where the conditional I added for the checkbox exists to strip the domain from the url. 

It then closes the modal and calls callback() passing in the amended text. I can only assume callback() is where it starts to head towards the server side of things.

dandwal commented on assets/florence/pagedown/Markdown.Editor.js:
```diff
@@ -1113,10 +1114,12 @@
                 if (!/^(?:https?|ftp):\/\//.test(text))
                     text = 'http://' + text;
+                if (check.checked == true)
+                    text = text.replace(/.*\/\/[^\/]*/, '')
```
This regex seems greedy and should it be anchored?  - anchoring from 'start (`^`) will also protect against any URLs like: `https://foo.com//bar` (where the `//bar` onwards will be considered a new match and thus all deleted).

What I read this to mean: As many of any characters followed by `//` followed by as many characters except `/`
It is good practice (although not done everywhere to comment what the goal of the regex is above the regex line.

JoshSinghONS replied:
So I will throw my hands in the air and admit I am no Regex Ninja so I pretty much googled "Regex that removes domain from url".

Happy to add the comment but unsure if you want me to change the Regex lol.

dandwal replied:
No worries, I think but maybe worth testing, it would be slightly better to change to:
`/^.*\/\/[^\/]*/, ''` - just the 'start of line anchor `^` at the beginning.

JoshSinghONS replied:
Done

dandwal commented on assets/florence/pagedown/Markdown.Editor.js:
```diff
@@ -1182,9 +1203,18 @@
+            style.fontSize = "0.8em";
+            style.fontWeight = "bold";
+
+            var div = doc.createElement("div");
```
Is this ES5 or under? If ES6 + can we use `let` / `const` rather than `var`?

JoshSinghONS replied:
No idea, how do I find out?

dandwal replied:
If it is only used in Florence then we know it will only be internal users using it on network devices, and currently, the internal internet browser is Chrome (Can check version as it will be the same as your Windows laptop as this is one of the on network devices) the version of Chrome supports ES6 so think we are safe to use let / const)

JoshSinghONS replied:
Done

dandwal reviewed (commented):
Can you add some steps as to where / how this can be tested please, along with what the aim of this PR is?
